lib/portage/package/ebuild/config.py

```diff
@@ -1087,6 +1088,24 @@ def _init_dirs(self):
 				writemsg("!!! %s\n" % str(e),
 					noiselevel=-1)
 
+	def _check_portage_homedir(self):
+		"""
+		Warn if portage home directory is missing or has incorrect ownership
+		"""
+		try:
+			stat = os.stat(os.path.expanduser("~portage"))
```

Comment:
You should not hardcode `portage`, but respect `portage.data._portage_username`.

`portage.data._portage_username` and `portage.data._portage_grpname` respect `PORTAGE_USERNAME` and `PORTAGE_GRPNAME` variables (which can be set in environment or `make.conf`).